lib/cloudware/models/concerns/deployment_tags.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Cloudware
+  module Models
+    module Concerns
+      module ModelTags
+        # NOTE: The TAG_TYPE must be set as a constant on the base model
+
+        def self.included(base)
+          base.extend(ClassMethods)
+        end
+
+        module ClassMethods
+          def flag
+            'TAG'
+          end
+
+          def build_from_context(context)
+            (context.results || {})
+                    .keys
+                    .map { |k| self.name_from_tag(k) }
+                    .uniq
+                    .reject { |n| n.nil? }
+                    .map do |name|
+              self.new(name: name, context: context)
+            end
+          end
+
+          def name_from_tag(tag)
+            regex = /\A.*(?=#{flag}.*\Z)/
+            regex.match(tag.to_s)&.to_a&.first
+          end
+
+          def tag_generator(name, tag)
+            :"#{name}#{flag}#{tag}"
+          end
+        end
+
+        attr_accessor :name, :context
+
+        def tags
+          (context.results || {}).each_with_object({}) do |(key, value), memo|
+            next unless (tag = extract_tag(key))
+            memo[tag] = value
+          end
+        end
+
+        def tag_generator(tag)
+          self.class.tag_generator(name, tag)
+        end
+
+        def extract_tag(key)
+          regex = /(?<=\A#{self.class.tag_generator(Regexp.escape(name), '')}).*/
```

Comment:
This regex needs a comment to explain what it actually matches pls.

Reply:
https://github.com/alces-software/cloudware/commit/90fc69eaa3ecfbc57b6ed4d299ed8902fb4e5b3e